staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go

```diff
@@ -384,3 +411,25 @@ func DefaultConsumeRequest(request rest.ResponseWrapper, out io.Writer) error {
 		}
 	}
 }
+
+type prefixingWriter struct {
+	prefix []byte
+	writer io.Writer
+}
+
+func (pw *prefixingWriter) Write(p []byte) (int, error) {
+	if len(p) == 0 {
+		return 0, nil
+	}
+
+	// Perform an "atomic" write of a prefix and p to make sure that it doesn't interleave
+	// sub-line when used concurrently with io.PipeWrite.
+	n, err := pw.writer.Write(append(pw.prefix, p...))
```

Comment:
I'd just 
```
return pw.writer.Write(append(pw.prefix, p...))
```

Reply:
From this function we must return number of bytes (`n`) written from `p`, where `n` must be `0 <= n <= len(p)`to meet the `io.Writer` interface requirements.

`return pw.writer.Write(append(pw.prefix, p...))` will be violating the `io.Writer` interface: it can return `n > len(p)` because we write `p` and `pw.prefix`. Other packages/functions will misbehave if we do that (I had issues with `io.Copy` which compares number of bytes internally, for example).

Reply:
Roger that :+1: